lib/auth.coffee

```diff
@@ -280,9 +295,15 @@ getAuth = (deps, opts) ->
 	# });
 	###
 	exports.getEmail = (callback) ->
-		token.getEmail().then (email) ->
-			throw new errors.ResinNotLoggedIn() if not email?
-			return email
+		request.send
+			method: 'GET'
+			url: '/user/v1/whoami'
+			baseUrl: apiUrl
+		.get('body')
+		.then (body) ->
+			return body.email
+		.catch ->
+			throw new errors.ResinNotLoggedIn()
```

Comment:
We should refactor out an internal function for this code, since we're repeating it a few times. It'd be nice if each of these methods did the same thing, but only needed to contain something like:

```coffeescript
getUserDetails()
.get('email')
.asCallback(callback)
```

Reply:
That would actually be especially nice because then we could add caching inside `getUserDetails`, to make all these methods very cheap after the 1st lookup.

Looking at the code below I think that caching is going to be necessary for this PR, since code like `getAll` gets called very frequently (I think the dashboard polls it every 5 seconds for example), and it now makes twice as many HTTP requests.